.github/workflows/docs-gh-pages.yml

```diff
@@ -1,19 +1,39 @@
-name: Publish docs via GitHub Pages
+name: Build and Deploy Documentation
+
 on:
   push:
     branches:
-      - main
+      # - main
+      - 72-api-documentation-with-sphinx
 
 jobs:
-  build:
-    name: Deploy docs
+  build-and-deploy-docs:
     runs-on: ubuntu-latest
     steps:
-      - name: Checkout main
-        uses: actions/checkout@v2
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Fetch all history for .git-restore-mtime to work correctly
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.x'
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install .[docs]
+
+      - name: Build MkDocs site (general documentation)
```

Comment:
Would it be fine to have 2 separate GH Actions from existing marketplace or is the current solution better? If so, in what way?

Reply:
I think it can be done, but it's a bit more annoying because they have poor interoperability. 
- For mkdocs, I'm referring to https://github.com/marketplace/actions/deploy-mkdocs. It uses mkdocs gh-deploy under the hood, which creates a completely new gh-pages branch and deletes the old one, so you have to use it before the sphinx build.
- For sphinx, I'm referring to https://github.com/sphinx-notes/pages. There's no way of specifying the output directory of the built files, so it would be more annoying to deal with the finished build paths.
Also this way I can specify requirements in the pyproject.toml instead of creating separate requirements.txt for mkdocs and sphinx and it's a bit faster because it doesn't have any extra features wrt GH Actions.

Reply:
OK, it's fine then.